autoware_plugin/src/autoware_plugin.cpp

```diff
@@ -48,12 +55,45 @@ namespace autoware_plugin
         pnh_->param<double>("trajectory_point_spacing", trajectory_point_spacing_, 0.1);
     }
 
+
     void AutowarePlugin::run()
     {
         initialize();
         ros::CARMANodeHandle::spin();
     }
 
+
+
+    bool AutowarePlugin::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest &req, cav_srvs::PlanTrajectoryResponse &resp){
+        resp.trajectory_plan = trajectory_msg;
+        resp.related_maneuvers.push_back(0);
+        resp.maneuver_status.push_back(1);
+
+        return true;
+    }
+
+    bool AutowarePlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp){        
+
+        cav_msgs::Maneuver maneuver_msg;
+
+        maneuver_msg.type = cav_msgs::Maneuver::LANE_FOLLOWING;  
+
+
+        maneuver_msg.lane_following_maneuver.start_dist = 0.0;
+        maneuver_msg.lane_following_maneuver.start_speed = waypoints_list[0].twist.twist.linear.x;
+        maneuver_msg.lane_following_maneuver.start_time = ros::Time::now();
+
+
+        maneuver_msg.lane_following_maneuver.end_dist = 20.0;
+        maneuver_msg.lane_following_maneuver.end_speed = waypoints_list[-1].twist.twist.linear.x;
```

Comment:
I'm not convinced this valid. A negative index into an array would give you undefined behavior. It would be better, to do waypoints_list[waypoints_list.size() - 1] if you are trying to get the last element.

Reply:
modified